spec/requests/signups_spec.rb

```diff
@@ -91,58 +99,102 @@
     end
     let(:installation_id) { SecureRandom.hex }
 
-    let(:signup) { create(:signup) }
 
-    before do
-      allow(Signups::Create).to receive(:call)
-        .with(installation_id: installation_id)
-        .and_return(signup)
-    end
+    context 'when validations succeed' do
+      before do
+        allow(Signups::CreateForm).to receive(:new)
+          .with(params.merge(installation_id: installation_id))
+          .and_return(form)
 
-    it "invokes singups create service" do
-      expect(Signups::Create).to receive(:call)
-        .with(installation_id: installation_id)
+        allow(form).to receive(:submit).and_return(signup)
+      end
+      let(:form) { instance_double(Signups::CreateForm) }
+      let(:signup) { create(:signup) }
 
-      dispatch_request
-    end
+      it "invokes singups create form" do
+        allow(Signups::CreateForm).to receive(:new)
+          .with(params.merge(installation_id: installation_id))
+          .and_return(form)
 
-    it "returns http success" do
-      dispatch_request
+        expect(form).to receive(:submit)
 
-      expect(response).to have_http_status(:success)
-    end
+        dispatch_request
+      end
 
-    it "returns registered signup as JSON" do
-      dispatch_request
+      it "returns http success" do
+        dispatch_request
 
-      expect(response.body).to eq({ id: signup.incognia_signup_id }.to_json)
-    end
+        expect(response).to have_http_status(:success)
+      end
 
-    it_behaves_like 'handle Incognia API errors' do
-      let(:service) { Signups::Create }
+      it "returns registered signup as JSON" do
+        dispatch_request
+
+        expect(response.body).to eq({ id: signup.incognia_signup_id }.to_json)
+      end
+
+      it_behaves_like 'handle Incognia API errors' do
+        let(:service) { form }
+        let(:method) { :submit }
+      end
+
+      context 'when structured address is informed' do
+        before { params.merge!(structured_address: address) }
+
+        let(:address) do
```

Comment:
What about using Faker to generate a valid address? Or maybe get from a signup generated by FactoryBot? :thinking:

Reply:
I think that when this code was created we didn't have those libraries, and this PR only moves this code to be inside a context block.

Fixed in this last PR: 94474d927ec87523b2d4450440d27a528d49905e